app/controllers/feedback.js

```diff
@@ -0,0 +1,94 @@
+const { isValidEmail, isValidWordCount } = require('../helpers/validation')
+
+exports.newFeedback_get = async (req, res) => {
+  const { feedback } = req.session.data
+  res.render('feedback/edit', {
+    feedback,
+    actions: {
+      back: '/',
+      cancel: '/',
+      save: '/feedback'
+    }
+  })
+}
+
+exports.newFeedback_post = async (req, res) => {
+  const { feedback } = req.session.data
+  const errors = []
+
+  const wordCount = 200
+
+  if (!feedback?.satisfaction) {
+    const error = {}
+    error.fieldName = 'satisfaction'
+    error.href = '#feedback-satisfaction'
+    error.text = 'Select how you feel about this service'
+    errors.push(error)
+  }
+
+  if (!feedback?.details.length) {
+    const error = {}
+    error.fieldName = 'details'
+    error.href = '#feedback-details'
+    error.text = 'Enter details about how we could improve this service'
+    errors.push(error)
+  } else if (!isValidWordCount(feedback.details, wordCount)) {
+    const error = {}
+    error.fieldName = 'details'
+    error.href = '#feedback-details'
+    error.text = `Details must be ${wordCount} words or fewer`
+    errors.push(error)
+  }
+
+  if (feedback?.email && !isValidEmail(feedback.email)) {
```

Comment:
![image](https://github.com/user-attachments/assets/47ededee-93c4-4a35-b859-e18edf0908cf)

Reply:
The whole section is marked optional. If you enter a name, we should require an email address and vice versa. If you enter neither, you don't get an error. It's a little finicky, but we can improve it later if it's a problem.